src/main/java/org/broadinstitute/hellbender/engine/FeatureDataSource.java

```diff
@@ -253,6 +253,9 @@ public FeatureDataSource(final FeatureInput<T> featureInput, final int queryLook
         } else {
             throw new GATKException("Found a feature input that was neither GenomicsDB or a Tribble AbstractFeatureReader.  Input was " + featureInput.toString() + ".");
         }
+        if (!hasIndex && TribbleIndexedFeatureReader.hasBlockCompressedExtension(featureInput.getFeaturePath())) {
+            throw new GATKException("Found a feature input that was block compressed but lacked an index file.  Input was " + featureInput.toString() + ".");
```

Comment:
I think that this is a temporary fix, no? If so, maybe it is worth to point that this restriction might be removed in the future.

In addition, shouldn't it be an `UserException` if the input should be indexed by the user?